### ISSUE TITLE
assert skip_window

### ISSUE THREAD

**Issue by yzexeter** [thushv89/udacity_deeplearning_complete#1]
Hi, Thank you for your example. I recently tried your code. There might be some bugs but I have solved it. 

embeds = tf.concat(2,[embeds,tf.reshape(embedding_i,[emb_x,emb_y,1])])
here I think it would be embeds = tf.concat([embeds,tf.reshape(embedding_i,[emb_x,emb_y,1])], axis = 2)
loss = tf.reduce_mean(tf.nn.sampled_softmax_loss(softmax_weights, softmax_biases, avg_embed,
                       train_labels, num_sampled, vocabulary_size))
here I think we need to change the parameters order to
loss = tf.reduce_mean(tf.nn.sampled_softmax_loss(softmax_weights, softmax_biases, train_labels, 
                       avg_embed, num_sampled, vocabulary_size))

But I still do not understand why you choose to assert skip_window%2==1 at the beginning of generate_batch(). I also reviewed the example of tensorflow implementation. From my understanding, the assertion here is to check if the window size is odd or even, which forces the size to be even otherwise raising an Error. I wonder if this is necessary?

**Comment by thushv89** [comment]
Hi Sorry about the delayed response. And thanks for pointing this out.

The code has been updated to fix the problems with [code]tf.nn.sampled_softmax_loss[/code]

And I've updated the repo and removed the unnecessary assert statement.
